mongodbatlas/resource_mongodbatlas_federated_settings_identity_provider.go

```diff
@@ -0,0 +1,265 @@
+package mongodbatlas
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/spf13/cast"
+)
+
+func resourceMongoDBAtlasFederatedSettingsIdentityProvider() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMongoDBAtlasFederatedSettingsIdentityProviderRead,
+		ReadContext:   resourceMongoDBAtlasFederatedSettingsIdentityProviderRead,
+		UpdateContext: resourceMongoDBAtlasFederatedSettingsIdentityProviderUpdate,
+		DeleteContext: resourceMongoDBAtlasFederatedSettingsIdentityProviderDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMongoDBAtlasFederatedSettingsIdentityProviderImportState,
+		},
+		Schema: map[string]*schema.Schema{
+			"federation_settings_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"issuer_uri": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"request_binding": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"response_signature_algorithm": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"associated_domains": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"sso_debug_enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"sso_url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"idp_id": {
```

Comment:
So since this has to be imported first we will already have this, correct?

Reply:
yes so we could have it as calculated field